src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.util.Arrays;
+import duke.task.Task;
+
+/*
+ * Deals with making sense of the user command
+ */
+public class Parser {
+
+    /**
+     * Runs the input command given by the user and write into file if list has changed
+     * 
+     * @param tasks Arraylist containing task objects
+     * @param storage Storage class that manages save and loading
+     * @param words User input
+     * @return true if command is bye, otherwise return false
+     * @throws IndexOutOfBoundsException When index given is larger than the taskList
+     * @throws NumberFormatException When a string is given instead of a number for some commmands
+     * @throws InvalidCommandException When user inputs an invalid command
+     * @throws EmptyDescriptionException When user inputs an empty description
+     */
+    public boolean runCommand(TaskList tasks, Storage storage, String[] words) throws IndexOutOfBoundsException,
+            NumberFormatException, InvalidCommandException, EmptyDescriptionException {
+
+        boolean hasTaskChanged = false;
+        words[0] = words[0].toUpperCase();
+        switch(words[0]) {
+        case "BYE":
```

Comment:
Would it be better to enumerate the different possible inputs so it's easier to keep track of them and can easily be used elsewhere too?